src/dotnet/CarbonAware.Aggregators/src/CarbonAware/SciScoreAggregator.cs

```diff
@@ -0,0 +1,53 @@
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using Microsoft.Extensions.Logging;
+using System.Collections;
+using System.Globalization;
+
+namespace CarbonAware.Aggregators.SciScore
+{
+    public class SciScoreAggregator : ISciScoreAggregator
+    {
+        private readonly ILogger<SciScoreAggregator> _logger;
+
+        private readonly ICarbonIntensityDataSource _carbonIntensityDataSource;
+
+        public SciScoreAggregator(ILogger<SciScoreAggregator> logger, ICarbonIntensityDataSource carbonIntensityDataSource)
+        {
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._carbonIntensityDataSource = carbonIntensityDataSource;
+        }
+
+        /// <inheritdoc />
+        public async Task<double> CalculateAverageCarbonIntensityAsync(Location location, string timeInterval)
```

Comment:
I'm a bit confused about how this (and this file as a whole) relates to the CarbonAware Aggregator:  https://github.com/microsoft/carbon-aware-sdk/blob/675076659d3055b63aeb11d24efe549257ecf1b2/src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs#L61

Reply:
Yes, that's the idea.  Right now it looks a little counter-intuitive, but once this aggregator has functions for how to get Energy and calculate a proper SCI score I think it will feel more like a good separation of concerns and less like duplicate code.

There are subtle differences as well.  A default `endDate` may make sense for the CarbonAware(Emissions)Aggregator to be consistent with the rest of that endpoint, whereas SCI score would certainly want to throw an exception.